spec/jsonapi/attributes_spec.rb

```diff
@@ -1,25 +1,41 @@
-require 'spec_helper'
-
 RSpec.describe JSONAPI::RSpec do
-  let(:doc) do
+  json_doc =
     {
       'attributes' => {
         'one' => 1,
         'two' => 2,
         'four' => 3
       }
     }
-  end
+
+  symbol_doc =
+    {
+      attributes: {
+        one: 1,
+        two: 2,
+        four: 3
+      }
+    }
 
   describe '#have_attribute' do
-    it { expect(doc).to have_attribute(:one) }
-    it { expect(doc).not_to have_attribute(:five) }
+    [json_doc, symbol_doc].each do |doc|
+      subject { doc }
+      before(:each) { RSpec.configuration.allow_symbolized_jsonapi = (doc == symbol_doc) }
+
+      it { is_expected.to have_attribute(:one) }
+      it { is_expected.not_to have_attribute(:five) }
+    end
```

Comment:
@sa73917 please try to avoid such loops in the tests, these are harder to read and maintain. You did a great job DRY-ing up the tests so all of these should be one-liners, but it helps a lot extending and maintaining the tests long-term.

Please maintain the explicit calls to `expect` instead of `is_expected`.

Reply:
Also, as I mentioned in the previous PR, we don't need to run both use-cases for every matcher. If it makes sense, let's add a unit-test to the `JSONAPI::RSpec.as_indifferent_hash` and since these all are calling it, it's more than enough to provide just one or two examples using string-based keys. The idea is to isolate, test and re-use component/parts of the code, and not carry over the tests for one component over and over. Let me know if this makes sense.